DataBiosphere/bard
# [DR-3027] Optionally turn off pushing events to mixPanel

https://broadworkbench.atlassian.net/browse/DR-3027

The Data Repo is sending too many events to MixPanel, so this PR adds the ability to turn off pushing events to mixPanel if a Bard event has the `pushToMixPanel` property set to `false`. If the property is not defined, Bard will send the event to MixPanel so this change does not impact existing services that use Bard.

Another change is that now all Bard event logs are sent to bigQuery (so even if a service opts out of mixPanel it is still possible to query the event logs). This was done by manually creating a log sink for dev and prod:
- Dev: The [bq-metrics](https://console.cloud.google.com/logs/router?project=terra-bard-dev) sink sends logs with `logName="projects/terra-bard-dev/logs/metrics"` to this [warehouse](console.cloud.google.com/bigquery?ws=!1m5!1m4!4m3!1sbroad-dsde-analytics-dev!2swarehouse!3smetrics) dataset (it looks like the automatically created table is called "metrics").
- Prod: The [bq-metrics](https://console.cloud.google.com/logs/router?authuser=3&project=terra-bard-prod) sink sends logs with `logName="projects/terra-bard-prod/logs/metrics"` to this [warehouse](console.cloud.google.com/bigquery?ws=!1m5!1m4!4m3!1sbroad-dsde-prod-analytics-dev!2swarehouse!3smetrics) dataset

**Note:** GCP creates a service account for each log sink that writes to the bigQuery dataset. I granted the service accounts bigQuery.dataEditor permissions on the destination datasets following: https://cloud.google.com/logging/docs/export/configure_export_v2#destination-paths

**Testing:**
1. Running Bard
- Follow the README instructions to run Bard locally. Change the port so that it does not conflict with TDR:
`GCP_PROJECT=terra-bard-dev GOOGLE_APPLICATION_CREDENTIALS=<path-to-key-file> PORT=8081 yarn run start-dev`

2. Running TDR
- Run this TDR branch locally https://github.com/DataBiosphere/jade-data-repo/pull/1471 and set the [bard base path](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/resources/application.properties#L124) to `http://localhost:8081`.
- Send a request to one of the GET endpoints
- Check the BigQuery table for the event (need to be logged in as your firecloud.org user). Note there can be a delay in BigQuery recording the `_PARTITIONTIME`.
